tests/Doc/QueryAggExample.cs

```diff
@@ -0,0 +1,304 @@
+                description = "An artsy,  retro-inspired bicycle that’s as functional as it is pretty: The ThrillCycle steel frame offers a smooth ride. A 9-speed drivetrain has enough gears for coasting in the city, but we wouldn’t suggest taking it to the mountains. Fenders protect you from mud, and a rear basket lets you transport groceries, flowers and books. The ThrillCycle comes with a limited lifetime warranty, so this little guy will last you long past graduation.",
+                condition = "refurbished"
+            }
+        };
+
+        for (var i = 0; i < bicycles.Length; i++)
+        {
+            db.JSON().Set($"bicycle:{i}", "$", bicycles[i]);
+        }
+        // HIDE_END
+
+
+        // STEP_START agg1
+        AggregationResult res1 = db.FT().Aggregate(
+            "idx:bicycle",
+            new AggregationRequest("@condition:{new}")
+                .Load(new FieldName("__key"), new FieldName("price"))
+                .Apply("@price - (@price * 0.1)", "discounted")
+        );
+        Console.WriteLine(res1.TotalResults);   // >>> 5
+
+        for (int i = 0; i < res1.TotalResults; i++)
+        {
+            Row res1Row = res1.GetRow(i);
+
+            Console.WriteLine(
+                $"Key: {res1Row["__key"]}, Price: {res1Row["price"]}, Discounted: {res1Row["discounted"]}"
+            );
+        }
+        // >>> Key: bicycle:0, Price: 270, Discounted: 243
+        // >>> Key: bicycle:5, Price: 810, Discounted: 729
+        // >>> Key: bicycle:6, Price: 2300, Discounted: 2070
+        // >>> Key: bicycle:7, Price: 430, Discounted: 387
+        // >>> Key: bicycle:8, Price: 1200, Discounted: 1080
+        // STEP_END
+
+        // Tests for 'agg1' step.
+        // REMOVE_START
+        Assert.Equal(5, res1.TotalResults);
+
+        Row test1Row = res1.GetRow(0);
+        Assert.Equal(
+            "Key: bicycle:0, Price: 270, Discounted: 243",
+            $"Key: {test1Row["__key"]}, Price: {test1Row["price"]}, Discounted: {test1Row["discounted"]}"
+        );
+
+        test1Row = res1.GetRow(1);
+        Assert.Equal(
+            "Key: bicycle:5, Price: 810, Discounted: 729",
+            $"Key: {test1Row["__key"]}, Price: {test1Row["price"]}, Discounted: {test1Row["discounted"]}"
+        );
+
+        test1Row = res1.GetRow(2);
+        Assert.Equal(
+            "Key: bicycle:6, Price: 2300, Discounted: 2070",
+            $"Key: {test1Row["__key"]}, Price: {test1Row["price"]}, Discounted: {test1Row["discounted"]}"
+        );
+
+        test1Row = res1.GetRow(3);
+        Assert.Equal(
+            "Key: bicycle:7, Price: 430, Discounted: 387",
+            $"Key: {test1Row["__key"]}, Price: {test1Row["price"]}, Discounted: {test1Row["discounted"]}"
+        );
+
+        test1Row = res1.GetRow(4);
+        Assert.Equal(
+            "Key: bicycle:8, Price: 1200, Discounted: 1080",
+            $"Key: {test1Row["__key"]}, Price: {test1Row["price"]}, Discounted: {test1Row["discounted"]}"
+        );
```

Comment:
not sure if Redis has a contract to have them in this exact same order in response.